flocker/dockerplugin/test/test_api.py

```diff
@@ -121,9 +137,86 @@ def test_create_with_opts(self):
                               Dataset(dataset_id=result[0].dataset_id,
                                       primary=self.NODE_A,
                                       maximum_size=int(DEFAULT_SIZE.to_Byte()),
-                                      metadata={u"name": name})]))
+                                      metadata={u"name": name,
+                                                u"clusterhq:flocker:profile":
+                                                unicode(profile)})]))
+        return d
+
+    def test_create_with_size(self):
+        """
+        Calling the ``/VolumerDriver.Create`` API with an ``Opts`` value
+        of "size=<somesize> in the request body JSON create a Volumes
+        with a given name and random size between 1-100G
+        """
+        name = random_name(self)
+        size = integers(min_value=1, max_value=75).example()
+        expression = volume_expression.example()
+        size_opt = "".join(str(size))+expression
+        d = self.assertResult(b"POST", b"/VolumeDriver.Create",
+                              {u"Name": name, 'Opts': {u"size": size_opt}},
+                              OK, {u"Err": None})
+
+        real_size = int(parse_num(size_opt).to_Byte())
+        d.addCallback(
+            lambda _: self.flocker_client.list_datasets_configuration())
+        d.addCallback(list)
+        d.addCallback(lambda result:
+                      self.assertItemsEqual(
+                          result, [
+                              Dataset(dataset_id=result[0].dataset_id,
+                                      primary=self.NODE_A,
+                                      maximum_size=real_size,
+                                      metadata={u"name": name,
+                                                u"maximum_size":
+                                                unicode(real_size)})]))
         return d
 
+    @given(expr=volume_expression,
+           size=integers(min_value=75, max_value=100))
+    def test_parsenum_size(self, expr, size):
+        """
+        Test sending different forms of size expressions
+        to parse_num and get expxted G size results.
```

Comment:
No need for "Test" in the docstring (see https://jml.io/test-docstrings/). Also, ideally try to say _what_ we expect, rather than "expected".